bon-macros/src/builder/builder_gen/mod.rs

```diff
@@ -233,6 +237,15 @@ impl BuilderGenCtx {
         let func = quote! {
             #(#docs)*
             #[inline(always)]
+            #[allow(
+                // This is intentional. We want the builder syntax to compile away
+                clippy::inline_always,
+                // We normalize `Self` references intentionally to simplify code generation
+                clippy::use_self,
+                // Let's keep it as non-const for now to avoid restricting ourselfves to only
+                // const operations.
+                clippy::missing_const_for_fn
```

Comment:
I'm curious whether `#[inline(always)]` will result in a kind of const operation anyway due to optimizations :thinking:

Having a const function might be better so maybe revisit this in the future?

Reply:
I'm 99% sure this will be come a `const fn` in the future. I'm just conservative about this, and I'd like to keep the posibility to use non-const operators here for `bon`, since I believe there are no uses cases for `const` builders, and besides just marking this fn `const` isn't enough. We need to have `IntoSet` trait const as well